src/Functions/array/arrayRemove.cpp

```diff
@@ -0,0 +1,65 @@
+#include <Functions/array/arrayRemove.h>
+#include <Functions/FunctionFactory.h>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_COLUMN;
+}
+
+ColumnPtr ArrayRemoveImpl::execute(const ColumnArray & array, ColumnPtr mapped)
+{
+    const ColumnUInt8 * column_filter = typeid_cast<const ColumnUInt8 *>(&*mapped);
+
+    if (!column_filter)
+    {
+        const auto * column_filter_const = checkAndGetColumnConst<ColumnUInt8>(&*mapped);
+
+        if (!column_filter_const)
+            throw Exception(ErrorCodes::ILLEGAL_COLUMN, "Unexpected type of filter column");
+
+        if (column_filter_const->getValue<UInt8>())
+            return array.clone();
+        else
+            return ColumnArray::create(
+                array.getDataPtr()->cloneEmpty(),
+                ColumnArray::ColumnOffsets::create(array.size(), 0));
+    }
+
+    const IColumn::Filter & filter = column_filter->getData();
+    IColumn::Filter negated_filter(filter.size()); // Create a new filter with the negated values, handling NULLs separately.
+
+    for (size_t i = 0; i < filter.size(); ++i)
```

Comment:
size is calculated in a loop

Reply:
the` for loop` isn't calculating the size, it is creating the resultant array with the negative offsets.
i tried out a lot of things and spoke to @nickitat too, this was the only method which worked :(